shaq_overflow/app/controllers/users_controller.rb

```diff
@@ -0,0 +1,62 @@
+class UsersController < ApplicationController
+  before_action :set_user, only: [:show, :edit, :update, :destroy]
+  before_action :require_login, except: [:new, :create]
+
+  def new
+    @user = User.new
+    render :new, layout: false
+  end
+
+  def create
+    @user = User.new(user_params)
+    respond_to do |format|
+      if @user.save
+        set_session
+        format.html { redirect_to @user, notice: 'User Created' }
+        format.json { render :user, status: :created, location: @user }
+      else
+        format.html { redirect_to :new_user }
+        format.json { render json: @user.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def show
+  end
+
+  def edit
+  end
+
+  def update
+    respond_to do |format|
+      if @user.update(user_params)
+        format.html { redirect_to @user, notice: 'User Updated' }
+        format.json { render :show, status: :updated, location: @user }
+      else
+        format.html { redirect to :edit }
+        format.json { render json: @user.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
+    session.clear if @user == current_user
+    @user.destroy
+    respond_to do |format|
+      format.html { redirect_to login }
+      format.json { head :no_content }
```

Comment:
Are you actually destroying users by AJAX? If you aren't but just thought you might be in future it's better not to implement. Think about YAGNI.